pkg/recommendation/capacity.go

```diff
@@ -0,0 +1,286 @@
+// and count for that capacity
+type capacityCount map[int64]int64
+
+// getOrCreate returns count of a given capacity. If key not
+// found then it adds that key in map and assign with 0
+func (cc capacityCount) getOrCreate(key int64) int64 {
+	value, found := cc[key]
+	if !found {
+		value = 0
+		cc[key] = value
+	}
+	return value
+}
+
+// update takes key and value as argument and update that map
+// with key value pair.
+func (cc capacityCount) update(key, value int64) {
+	cc[key] = value
+}
+
+// getMinMaxCapacityForRaidGroupConfig returns takes raidgroup config as
+// an input and returns min and max capacity in resource.Quantity format
+func (cc capacityCount) getMinMaxCapacityForRaidGroupConfig(
+	raidConfig types.RaidGroupConfig) (min, max resource.Quantity) {
+	max = resource.Quantity{}
+	max = resource.Quantity{}
+
+	// Ideally caller will check raid group config is valid not
+	// If invalid raid config passed then it will not return error.
+	// It only logs the error and return empty struct
+	if err := raidConfig.Validate(); err != nil {
+		// TODO log the error.
+		return
+	}
+
+	for capacity, count := range cc {
+		// If device count is less than group device count
+		// then skip those block devices.
+		if count < raidConfig.GroupDeviceCount {
+			continue
+		}
+
+		// Get newMin capacity from block device capacity and count.
+		// If min capacity in response 0 or greater than newMin then
+		// update the min to newMin.
+		newMin, err := resource.ParseQuantity(fmt.Sprintf("%d",
+			capacity*raidConfig.GetDataDeviceCount()))
+		if err != nil {
+			// TODO log the error
+			continue
+		}
+		if min.IsZero() || min.Cmp(newMin) > 0 {
+			min = newMin
+		}
+
+		// noOfRaidGroup raid group count that can be formed.
+		// ie - one node has 5 block devices of 100GB and pool type
+		// is mirror then noOfRaidGroup will be 2.
+		noOfRaidGroup := count / raidConfig.GroupDeviceCount
+
+		// Get newMax capacity from block device capacity and count.
+		// If max capacity in response 0 or less than newMax then
+		// update the max to new max.
+		newMax, err := resource.ParseQuantity(fmt.Sprintf("%d",
+			capacity*noOfRaidGroup*raidConfig.GetDataDeviceCount()))
+		if err != nil {
+			// TODO log the error
+			continue
+		}
+		if max.IsZero() || max.Cmp(newMax) < 0 {
+			max = newMax
+		}
+	}
+
+	return
+}
+
+// nodeCapacityCount is a typed map contains node name and capacitycount
+type nodeCapacityCount map[string]capacityCount
+
+// getOrCreate returns capacityCount of a given node. If key not
+// found then it adds that key in map and assign with default value
+func (ncc nodeCapacityCount) getOrCreate(key string) capacityCount {
+	value, found := ncc[key]
+	if !found {
+		value = capacityCount{}
+		ncc[key] = value
+	}
+	return value
+}
+
+// update takes key and value as argument and update that map
+// with key value pair.
+func (ncc nodeCapacityCount) update(key string, value capacityCount) {
```

Comment:
updateCapacityInventory(node, inv)

Reply:
I think update is good else there will be a repetition of word.

Reply:
IMO what is being updated is not known when method name is update